python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -986,24 +977,21 @@ class RandomForestClassifier(Base):
                             then please use the CPU based predict by \
                             setting predict_model = 'CPU'")
 
-        elif self.num_classes > 2:
-            raise NotImplementedError("Predict_proba for multi-class "
-                                      "classification models is currently not "
-                                      "implemented. Please check cuml issue "
-                                      "#1679 for more information.")
+        if num_classes != self.num_classes and num_classes != 'auto':
+            raise NotImplementedError("limiting num_classes for predict_proba \
+                                      is not implemented")
         preds_proba = \
             self._predict_model_on_gpu(X, output_class=output_class,
                                        threshold=threshold,
                                        algo=algo,
-                                       num_classes=num_classes,
                                        convert_dtype=convert_dtype,
                                        fil_sparse_format=fil_sparse_format,
                                        predict_proba=True)
 
         return preds_proba
 
     def score(self, X, y, threshold=0.5,
-              algo='auto', num_classes=2, predict_model="GPU",
+              algo='auto', num_classes='auto', predict_model="GPU",
```

Comment:
I think we should remove the option of accepting `num_classes` from the user here as well.